app/gui/src/presenter/graph/state.rs

```diff
@@ -454,14 +454,19 @@ impl<'a> ControllerChange<'a> {
         let new_displayed_expr = node_view::Expression {
             pattern:             node.info.pattern().map(|t| t.repr()),
             code:                node.info.expression().repr().into(),
-            whole_expression_id: node.info.expression().id,
+            whole_expression_id: Some(node.info.id()),
```

Comment:
When touching the code, I noticed that `whole_expression_id` is in fact the id of the displayed portion of the expression only. I'm not sure if my fix is correct, but `whole_expression_id` seems to be unused in code outside tests.